h2o-py/h2o/h2o.py

```diff
@@ -1257,6 +1259,102 @@ def make_metrics(predicted, actual, domain=None, distribution=None):
+    :param name: file name to find
+    :param path: folder/directory name with full path
+    :return: Successful path name or None if failed
+    """
+    for root, dirs, files in os.walk(path):
+        if name in files:
+            return os.path.join(root, name)
+
+def check_java_process_and_version():
+    """
+    Helper function for predict_json() to check java version in the machine
+    :return: True/False if support java is available
+    """
+    result = subprocess.call(["java", "-version"])
+    if result is not 0:
+        raise RuntimeError("Java process not found, exiting!!")
+
+    output = subprocess.check_output(["java", "-version"], stderr=subprocess.STDOUT)
+    #output = subprocess.Popen(["java", "-version"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    #print output.communicate()
+    #print output.wait()
+    result = "0.0"
+    supported = False
+    if output.find("1.6.") is not -1:
+        result = "1.6"
+    elif output.find("1.7.") is not -1:
+        result = "1.7"
+        supported = True
+    elif output.find("1.8.") is not -1:
+        result = "1.8"
+        supported = True
+    return supported
+
+def predict_json(dir_path_to_genmodel_and_mojo, mojo_model_name, json_data, show_debug=False):
+    """
+    MOJO scoring function to take a pandas data frame as json string and use MOJO model as zip file to score
+
+    :param dir_path_to_genmodel_and_mojo: This is the local file system folder name where both h2o-genmodel.jar and MOJO zip file is stored.
+    :param mojo_model_name: This is the MOJO model file name which is download after model was build using H2O
+    :param json_data:  convert pandas frame to json (pd.to_json) and pass here
+    :param show_debug: True/False - Default ( Use True to see the full Java command line)
+    :return: score as json values
+    """
+    # Checking java
+    if not check_java_process_and_version():
+        raise RuntimeError("Error: Java version is not supported!!")
+
+    # Check both genmodel and pojo is available in the same path
+    # Try assert?
+    # assert os.path.exists(dir_path_to_genmodel_and_mojo + "/" + mojo_model_name)
+    mojo_test = find_file(mojo_model_name, dir_path_to_genmodel_and_mojo)
+    if mojo_test is None:
+        raise RuntimeError("Error: MOJO model is not available in " + dir_path_to_genmodel_and_mojo + " folder.")
+
+
+    # assert os.path.exists(dir_path_to_genmodel_and_mojo + "/" + gen_model_file_name)
+    genmodel_test = find_file(gen_model_file_name, dir_path_to_genmodel_and_mojo)
+    if genmodel_test is None:
+        raise RuntimeError("Error:" + gen_model_file_name +  " is not available in " + dir_path_to_genmodel_and_mojo + " folder.")
+
+        # dir_path_to_genmodel_and_mojo > must start and ends with "/" or add to it
+    gen_model_arg = ".:"
+
+    temp_dir_path =  dir_path_to_genmodel_and_mojo
+
+    if (dir_path_to_genmodel_and_mojo.endswith("/")):
```

Comment:
We should probably use existing methods form os module (os.path.sep.join, os.path.abspath)

Reply:
This line is just check to make sure if the user given path ends with "/" or not.